### ISSUE TITLE
Passing arguments into toolings with multiple commands no longer works

### ISSUE THREAD

**Issue by jklmnop** [lando/lando#2549]
After upgrading to v3.0.9 for MacOSX (10.14.6), my tooling broke. I have a tool for phinx that changes the directory in the container so I can execute phinx commands from anywhere in the project on my local machine. v3.0.9 has broken that functionality.

Previous version passed any additional arguments to the end of whatever my `cmd` was.

Below is a simple example that should change the dir and echo whatever you pass after executing the tooling in the terminal.

```yaml
// .lando.yml

tooling:
  cdandecho:
    service: appserver
    cmd: cd /app/some/dir && echo
```

```bash
$ lando cdandecho hello
// echoes nothing, expected hello
```

**Comment by dustinleblanc** [comment]
Hmm, this might be related to #2546

**Comment by certainlyakey** [comment]
Have the same problem. With tooling config like: 

```
tooling:
  listfolder:
    service: node
    cmd: 
      - cd $LANDO_MOUNT/some/specific/folder && ls 
```

and this command executed: `lando listfolder -ls` Lando displays a result of `ls` execution instead of `ls -al`: 

```
common_config.json  composer.lock
```

instead of this: 

```
-rw-r--r--    1 node node    278 Jul 30 14:40 common_config.json
-rw-r--r--    1 node node    125 Jul 25 09:45 composer.lock
```

**Comment by miiimooo** [comment]
Same here. I have this in `tooling:` for composer to avoid running out of memory but after upgrading to 3.0.9 any `lando composer <command>` displays the composer help pages only

```
  composer:
    service: appserver
    cmd: "cd $LANDO_MOUNT && php -d memory_limit=-1 /usr/local/bin/composer"
```

**Comment by dustinleblanc** [comment]
@miiimooo Composer should already be using unlimited memory in all of our recipes afaik, so you shouldn't have to do that.

It does sound like custom tooling commands changed though, so it does sound like we have a bug to fix

**Comment by joshuami** [comment]
I can confirm this bug started with v3.0.9. We have a similar command in our recipe that runs yarn commands within a Drupal theme:
```
  yarn:
    description: CUSTOM Run yarn commands
    service: node
    cmd: cd /app/web/themes/custom/cloudy && yarn
```

We have a similar command in our appserver service that works fine for using Drush that is unaffected, but it also uses the root value instead of a cd in to the directory.
```
  drush:
    description: CUSTOM Run drush commands
    service: appserver
    cmd: drush --root=/app/web
```

Interestingly, the node commands actually run on a `lando start` or `lando rebuild` where they are part of a run:

```
    run:
      - echo "Installing Node.js dependencies..."
      - cd /app/web/themes/custom/cloudy && yarn install --ignore-engines --frozen-lockfile
```

At first I thought what was lost was the ability to `cd` within a tooling `cmd:`, but if I add a `pwd` to the chain of commands, it prints the correct directory:

```
  yarn:
    description: CUSTOM Run yarn commands
    service: node
    cmd: echo "Trying" && cd /app/web/themes/custom/cloudy && pwd && yarn
```

Running `lando yarn` yields:
```
Trying
/app/web/themes/custom/cloudy
yarn install v1.22.4
[1/4] Resolving packages...
success Already up-to-date.
Done in 1.23s.
```

So it is running yarn, just not in the correct directory to detect the package.json. If I remove the `cd` command from the mix and just `cd` into the theme before running yarn. I can get everything to build. I can also get specific yarn commands to work correctly if a create a tooling command like `yarnbuild` that runs `cd /app/web/themes/custom/cloudy && yarn run build`. I think what was lost was the ability to create a tooling command that acts kinda like a bash alias. 

I'm pretty sure this happened in @pirog 's work to improve tooling commands that was merged as a part of v3.0.9: https://github.com/lando/lando/commit/2fa09dc041599020287c5bc8a67a211c6f5566c7.

He would probably best be able to tell us if this is something that could get resolved in a future release or if that is just the way tooling will work going forward. Personally, I like the ability to be able to use a tooling command like a fancy alias, but I get why it would make parsing those arguments challenging.

**Comment by garrettw** [comment]
I have this issue as well, with the following tooling command:
```yaml
  phpunit:
    service: appserver
    cmd: cd api && vendor/bin/phpunit -v --cache-result --order-by=defects,default --stop-on-failure
```

If I do `lando phpunit --filter ClassName`, nothing gets filtered, as if it didn't even pass in the extra args.

**Comment by hailwood** [comment]
This one's starting to bite. I've now got commands for most of the common usecases I.e.

```yml
gulp:
  service: node-cli
  description: "Build and watch"
  cmd: cd $LANDO_MOUNT && cd httpdocs/themes/default && ./node_modules/.bin/gulp
gulp-build:
  service: node-cli
  description: "Build the theme"
  cmd: cd $LANDO_MOUNT && cd httpdocs/themes/default && ./node_modules/.bin/gulp build
gulp-clean:
  service: node-cli
  description: "Clean the output"
  cmd: cd $LANDO_MOUNT && cd httpdocs/themes/default && ./node_modules/.bin/gulp clean
```

When I used to be able to just do `lando gulp clean` or `lando gulp`

**Comment by pirog** [comment]
This is an example of bad tooling design. As you note this is inherently not scalable. 

```yaml
gulp:
  service: node-cli
  description: "Build and watch"
  cmd: cd $LANDO_MOUNT && cd httpdocs/themes/default && ./node_modules/.bin/gulp
gulp-build:
  service: node-cli
  description: "Build the theme"
  cmd: cd $LANDO_MOUNT && cd httpdocs/themes/default && ./node_modules/.bin/gulp build
gulp-clean:
  service: node-cli
  description: "Clean the output"
  cmd: cd $LANDO_MOUNT && cd httpdocs/themes/default && ./node_modules/.bin/gulp clean
```

What you want to do is just this

```cmd
gulp:
  service: node-cli
  cmd: /path/to/gulp/node_modules/.bin/gulp
```

and then

```bash
cd /to/where/i/want/to/gulp
gulp some-thing 
```

**Comment by hailwood** [comment]
I wasn't aware you could invoke tooling from somewhere that wasn't the root directory. 

That makes it much easier 👍

**Comment by jklmnop** [comment]
While [this ](https://github.com/lando/lando/issues/2549#issuecomment-692665045)would allow the tooling to work as a workaround, the issue still persists if you include the `cd` into the tooling `cmd`. This is very handy for many tools (gulp, grunt, drush, phinx, etc) that require execution in a specific directory and are not likely to change per project. Based on their example, I believe the following is what @hailwood is trying to achieve. 

This approach used to allow the tooling to be executed successfully from any directory within the project before 3.0.9.

```yaml
gulp:
  service: node-cli
  cmd: cd /to/where/i/want/to/gulp && /path/to/gulp/node_modules/.bin/gulp
```

```bash
// this won't work and will probably just give you the gulp help 
// because "some-thing" is not passed to the gulp command
$ lando gulp some-thing
```

**Comment by joshuami** [comment]
@pirog, I agree it doesn't make sense to have a tooling entry per command on that tool, but I do think there is a case for being able to set a pass-through directory for the tooling to be run.

For almost two years of using Lando, we've been able create tooling that passes a command to a specific folder in our project. Up until 3.0.9 using `cd` in the tooling worked for telling the tooling what we intended.

The use case is that we are running a terminal window in our code editor (in this case VS Code, but it could be any) and we only really use that terminal to pass commands at the root of our project. Basically, every `lando` command we run, we tell it where that command where it should run in that particular project. This really cuts down on the documentation for the team as we don't have to tell them to `cd` into the correct directory before running the command.

Today we found a pretty slick work around for `yarn` that works similar to how we use `drush` in our Drupal projects.

```
  yarn:
    description: CUSTOM Run yarn commands
    service: node
    cmd: yarn --cwd /app/web/themes/custom/$theme
```

(Where `$theme` is the name of our theme.)

This works as it had worked before the 3.0.9 upgrade.

Interestingly, a similar command for `npm` (different project, different theme approach) works, but with a weird side effect.

```
  npm:
    description: Run npm commands
    service: node
    cmd: npm -C /app/web/themes/custom/$theme
```

Adding the -C (shorthand for --prefix) executes our `npm` commands from within the correct directly just fine, but it also seems to clear the command prompt for the command for some reason I don't immediately see.

To sum up, there is a pretty solid use case for being able to use Lando's tooling to run all your CI/CD and build/watch commands from the command prompt in the root of your project. If you aren't constantly editing files from within the terminal, I don't know why you wouldn't want to just have your project's tooling to run from the root directory and work against files like a package.json that might be in a sub-directory per your chosen platform's convention.

Just a note to those following this thread, it looks like `--pwd` is also valid for telling gulp where to look for its file. (Haven't tried it as we aren't currently using Gulp, but it might be worth a look to unblock upgrading to the latest Lando version.)

**Comment by pirog** [comment]
@joshuami the issue is still open so we agree this is a bug and that this should be supported, we are still trying to figure out the best way to fix this regression without causing even more headaches

**Comment by jklmnop** [comment]
@pirog while I’d love for this to work as it used to, I understand it may be difficult to return the functionality to its previous state.

Since many of the use cases here seem to be changing into a directory before using a executable with arguments, maybe there could be a yaml key for specifying the target directory for executing the tooling?

It’s not really solving the full extent of the issue but it solves a common use case. Just a thought.

**Comment by dustinleblanc** [comment]
I don't the specifics of the complexity of that kind of change, but generally, as a user interface, having a `working_dir` key or something to specify a custom directory would be cool.

**Comment by pirog** [comment]
I think that would also be an awesome feature.

**Comment by garrettw** [comment]
Any progress on this?

**Comment by pirog** [comment]
I think the short term solution is to add support for a `working_dir` option. Longer term i think this probably needs to be addressed in a next major version sort of way.

**Comment by pirog** [comment]
https://docs.lando.dev/config/tooling.html#current-working-directory this is now a thing

**Comment by jklmnop** [comment]
Thanks so much, @pirog! Can't wait to update!

**Comment by jonathanjfshaw** [comment]
This still persists as a bug AFAIK

**Comment by Jalliuz** [comment]
Using '&&' and then arguments is the problem
Solution is to use the pipe `|` and use `"$@" ` to pass arguments

Wrong (using &&):
````
tooling:
    run-tests:
        service: appserver
        cmd: cd /app/some-folder && export NO_DB_RESET=1 && bin/simple-phpunit
````

Right (using | and "$@"), plus also move the "cd" command to a dedicated "dir" key
````
tooling:
    run-tests:
        service: appserver
        dir: /app/some-folder
        cmd: |
            export NO_DB_RESET=1
            bin/simple-phpunit "$@"
````

After this change I can run my tooling with an argument

`lando run-tests tests/UtilityTests/NumberUtilsTest.php`
